tests/uv/mlflow/test_reporter.py

```diff
@@ -201,3 +204,76 @@ def test_report():
         cur_step = s['step']
         for k, v in s['m'].items():
           assert metric_data[k][cur_step] == v
+
+
+def test_pubsub_report(monkeypatch):
+
+  with tempfile.TemporaryDirectory() as tmpdir:
+    mlf.set_tracking_uri(f'file:{tmpdir}/foo')
+    _reset_experiment()
+
+    # we need to access the run via a client to inspect most params/tags/metrics
+    client = mlf.tracking.MlflowClient()
+
+    class MockClient():
+
+      def __init__(self, batch_settings=(), publisher_options=(), **kwargs):
+        pass
+
+      def publish(self, topic: str, msg: bytes):
+        d = json.loads(msg.decode('utf-8'))
+        run_id = d['run_id']
+        metrics = [Metric(**x) for x in d['metrics']]
+        print(f'{d}')
+        client.log_batch(run_id=run_id, metrics=metrics)
+
+    monkeypatch.setattr(google.cloud.pubsub_v1, 'PublisherClient', MockClient)
+
+    mlflow_cfg = {
+        'experiment_name': 'foo',
+        'run_name': 'bar',
+        'artifact_location': '/foo/bar',
+    }
+
+    with uv.start_run(**mlflow_cfg) as active_run, uv.active_reporter(
+        MLFlowReporter(pubsub_topic='test')) as r:
+      assert r is not None
+
+      steps = [{
```

Comment:
if you have two events, I recommend calling `report` or `report_all` directly, since it's easier to read